pymc/aesaraf.py

```diff
@@ -537,23 +537,64 @@ def join_nonshared_inputs(
     point: Dict[str, np.ndarray],
     xs: List[TensorVariable],
     vars: List[TensorVariable],
-    shared,
+    shared: Dict[TensorVariable, TensorSharedVariable],
     make_shared: bool = False,
-):
+) -> Tuple[List[TensorVariable], TensorVariable]:
     """
-    Takes a list of Aesara Variables and joins their non shared inputs into a single input.
+    Takes a list of TensorVariables and joins their non shared inputs into a single input.
 
     Parameters
     ----------
-    point: a sample point
-    xs: list of Aesara tensors
-    vars: list of variables to join
+    point: a sample point associated with a model and variable inputs
+    xs: list of TensorVariable to replace subgraphs with vars and shared
+    vars: list of TensorVariable to reshape, join, and return as inarray
+    shared: dict of TensorVariable and their associated TensorSharedVariable in
+        subgraph replacement
+    make_shared: bool flag to use point argument to build subgraphs of xs_special
 
     Returns
     -------
-    tensors, inarray
-    tensors: list of same tensors but with inarray as input
-    inarray: vector of inputs
+    xs_special, inarray
+    xs_special: list of same xs TensorVariables but with inarray and shared in subgraphs
+    inarray: TensorVariable built from vars argument
+
+    Examples
+    --------
+    Join all model variables with none being shared with respect to model logp.
+
+    .. code-block:: python
+
+        initial_point = model.initial_point()
+        all_model_variables = pm.aesaraf.inputvars(model.value_vars)
+        # Will be empty here since nothing to replace
+        shared = pm.aesara.make_shared_replacements(
+            point=initial_point,
+            vars=all_model_variables,
+            model=model,
+        )
+        [logp0], inarray0 = pm.aesara.join_nonshared_inputs(
+            point=initial_point,
+            xs=[model.logp()],
+            vars=all_model_variables,
+            shared=shared,
+        )
```

Comment:
Let's simplify this example:

```suggestion
    Join the input value variables of a model logp.

    .. code-block:: python

        import pymc as pm

        with pm.Model() as model:
            mu_pop = pm.Normal("mu_pop")
            sigma_pop = pm.HalfNormal("sigma_pop")
            mu = pm.Normal("mu", mu_pop, sigma_pop, shape=(3,))
            y = pm.Normal("y", mu, 1.0, observed=[0, 1, 2])

        print(model.compile_logp()({"mu_pop": 0, "sigma_pop_log__": 1, "mu": [0, 1, 2]}))  # -12.691227342634292

        initial_point = model.initial_point()
        inputs = model.value_vars

        [logp], joined_inputs = join_nonshared_inputs(
            point=initial_point,
            xs=[model.logp()],
            vars=inputs,
            shared=[],
        )

        print(logp.eval({joined_inputs:[0, 1, 0, 1, 2]}))  # -12.691227342634292

```